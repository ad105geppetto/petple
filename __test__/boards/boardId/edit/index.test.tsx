import "@testing-library/jest-dom";
import { MockedProvider } from "@apollo/client/testing";
import React, { useState as useStateMock } from "react";
import {
  act,
  fireEvent,
  render,
  screen,
  waitFor,
} from "@testing-library/react";
import { useRouter } from "next/router";
import { RecoilRoot } from "recoil";
import BoardWrite from "../../../../src/components/units/board/write/BoardWrite.container";
import BoardWriteUI from "../../../../src/components/units/board/write/BoardWrite.presenter";
import { mocks, propsMock } from "./boardEdit.mocks";

jest.mock("next/router", () => ({
  useRouter: jest.fn(),
}));
(useRouter as jest.Mock).mockImplementation(() => ({
  push: jest.fn(),
  query: { boardId: "6409e1391182750028ee7891" },
}));

describe("자유게시판 수정 페이지 Unit Test & Integration Test", () => {
  it("게시판 수정 페이지 스냅샷 테스트", async () => {
    await act(async () => {
      const result = render(
        <MockedProvider mocks={mocks} addTypename={false}>
          <BoardWriteUI {...propsMock} />
        </MockedProvider>
      );

      expect(result.container).toMatchSnapshot();
    });
  });

  it("내용작성 후 '수정하기' 버튼 클릭시, 페이지 이동합니다.", async () => {
    const router = useRouter();
    const setContents = jest.fn();

    await act(async () => {
      render(
        <RecoilRoot>
          <MockedProvider mocks={mocks} addTypename={false}>
            <BoardWrite
              isEdit={true}
              router={router}
              contents={"update_test"}
              setContents={setContents}
            />
          </MockedProvider>
        </RecoilRoot>
      );
    });

    const password = screen.getByRole("password");
    fireEvent.change(password, { target: { value: "1234" } });
    const title = screen.getByRole("title");
    fireEvent.change(title, { target: { value: "update_test" } });

    const onClickList = screen.getByText("수정하기");
    fireEvent.click(onClickList);

    await waitFor(() => {
      expect(router.push).toHaveBeenCalled();
    });
  });
});
