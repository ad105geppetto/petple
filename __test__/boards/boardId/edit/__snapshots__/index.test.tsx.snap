// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`자유게시판 수정 페이지 Unit Test & Integration Test 게시판 수정 페이지 스냅샷 테스트 1`] = `
<div>
  
  <h2
    class="css-1iws72y"
  >
    게시물 
    수정
  </h2>
  <div
    class="css-1nu9z0v"
  >
    <div
      class="css-1p63ots"
    >
      <h4
        class="css-qmq5zt"
      >
        작성자
      </h4>
      <input
        class="css-4si2gt"
        role="writer"
        type="text"
        value="test"
      />
    </div>
    <div
      class="css-1p63ots"
    >
      <h4
        class="css-qmq5zt"
      >
        비밀번호
      </h4>
      <input
        class="css-4si2gt"
        role="password"
        type="password"
      />
    </div>
  </div>
  <div
    class="css-1p63ots"
  >
    <h4
      class="css-qmq5zt"
    >
      제목
    </h4>
    <input
      class="css-10fc8nt"
      role="title"
      type="text"
      value="test"
    />
  </div>
  <div
    class="css-sabcgf"
  >
    <h4
      class="css-qmq5zt"
    >
      내용
    </h4>
    <p>
      Loading ...
    </p>
  </div>
  <div
    class="css-iy9twh"
  >
    <h4
      class="css-qmq5zt"
    >
      사진 첨부
    </h4>
    <div
      class="css-2j4r9o"
    >
      <div
        class="css-ieayhg"
      >
        <div
          class="css-mg6t8m"
        >
          <div
            class="css-1kbdxi2"
          >
            +
          </div>
          <div
            class="css-1nx70zi"
          >
            Upload
          </div>
        </div>
        <input
          accept="image/jpeg, image/jpg, image/png"
          class="css-38lglc"
          type="file"
        />
      </div>
      <div
        class="css-ieayhg"
      >
        <div
          class="css-mg6t8m"
        >
          <div
            class="css-1kbdxi2"
          >
            +
          </div>
          <div
            class="css-1nx70zi"
          >
            Upload
          </div>
        </div>
        <input
          accept="image/jpeg, image/jpg, image/png"
          class="css-38lglc"
          type="file"
        />
      </div>
      <div
        class="css-ieayhg"
      >
        <div
          class="css-mg6t8m"
        >
          <div
            class="css-1kbdxi2"
          >
            +
          </div>
          <div
            class="css-1nx70zi"
          >
            Upload
          </div>
        </div>
        <input
          accept="image/jpeg, image/jpg, image/png"
          class="css-38lglc"
          type="file"
        />
      </div>
    </div>
  </div>
  <div
    class="css-1q2lbmy"
  >
    <div
      class="css-esuycf"
    >
      수정하기
    </div>
  </div>
</div>
`;
